tools/ci/tc/decision.py

```diff
@@ -249,6 +250,31 @@ def create_tc_task(event, task, taskgroup_id, depends_on_ids, env_extra=None):
     return task_id, task_data
 
 
+def create_sink_task(event, taskgroup_id, task_id_map):
+    depends_on_ids = list(map(lambda x: x[0], task_id_map.values()))
+
+    task_id = taskcluster.slugId()
+    task_data = {
+        "taskGroupId": taskgroup_id,
+        "created": taskcluster.fromNowJSON(""),
+        "deadline": taskcluster.fromNowJSON("24 hours"),
+        "provisionerId": "built-in",
+        "schedulerId": "taskcluster-github",
+        "workerType": "succeed",
+        "metadata": {
+            "name": "sink-task",
+            "description": "Sink task for all other tasks, indicates success",
+            "owner": get_owner(event),
+            "source": event["repository"]["clone_url"]
+        },
+        "payload": {},
+        "routes": ["checks"],
+        "dependencies": depends_on_ids,
+        "requires": "all-completed"
```

Comment:
> Sorry for this rather late design feedback

No worries, happy to make changes. I think your idea is a good one, to make us actually get a failure rather than an always-pending sink task (which is what happens when a sub-test fails currently, I believe). I'll look into this over the next few days.